inst/tutorials/tidyverse_ws/tidyverse_ws.Rmd

```diff
@@ -0,0 +1,1060 @@
+---
+title: "Introduction to the R tidyverse"
+author: "Gil J. B. Henriques, Kim Dill-McFarland and Kris Hong.<br>
+        A tutorial by [ECOSCOPE](http://ecoscope.ubc.ca/) at UBC."
+date: "version `r format(Sys.time(), '%B %d, %Y')`"
+description: In-depth description will follow.
+output: 
+  learnr::tutorial:
+      progressive: true
+      allow_skip: true
+runtime: shiny_prerendered
+---
+
+```{r setup, include = FALSE}
+# General learnr setup
+library(learnr)
+knitr::opts_chunk$set(echo = FALSE)
+library(educer)
+# Helper function to set path to images to "/images" etc.
+setup_resources()
+
+# Tutorial specific setup
+library(tidyverse) # Suite of packages for data manipulation and visualization
+library(lubridate) # Special formatting for dates within data
+library(cowplot) # Multi-panel figures
+library(readr)
+
+```
+
+## Introduction
+In this tutorial, we will learn to manipulate and plot real-world data using the [tidyverse](https://www.tidyverse.org/), a collection of R packages for data analysis and visualization.
+
+You will learn how to:
+
+- Create subsets of rows or columns from data frames using dplyr,
+- Change your data frames between wide and long formats,
+- Create various types of graphics,
+- Modify the various features of a graphic, and
+- Save your graphic in various formats
+
+This comprehensive introduction to the tidyverse is ideal for beginners but assumes some prior experience with R (e.g. as provided in our "Introduction to R and RStudio" tutorial).
+
+### How to follow this tutorial
+
+```{r child = system.file("resources/markdown", "how_to_follow_tutorials.Rmd", package = "educer")}
+```
+
+### Data description
+
+``` {r child = system.file("resources/markdown", "data_description.Rmd", package = "educer")}
+```
+
+During this tutorial, the data are accessible as a data frame called `geochemicals`. In the interactive code box below, use the function `head` to peak at the first few rows of this data frame:
+
+```{r geochemicals, exercise = TRUE}
+head(<name_of_the_data_frame>)
+```
+
+```{r geochemicals-solution}
+head(geochemicals)
+```
+
+
+### Loading packages
+
+The [tidyverse](https://www.tidyverse.org/) is a collection of R packages. Packages expand R's capabilities by providing additional functions. Some tidyverse packages that we will use in this tutorial are: dplyr (for data manipulation; pronounced *dee-plier*), tidyr (for rearranging data frames), and ggplot2 (for data visualization).  We will also use the packages lubridate (for working with dates) and cowplot (for multi-panel figures).
+
+When you work in your own projects, always load the packages you need at the top of your script. You can load each one of the tidyverse packages individually, but since they are often used together, you can also download and load the entire suite of tidyverse packages at once. In the future, you can use the code below to load all of the R packages we will use in this tutorial. 
+
```

Comment:
```suggestion
Alternatively, when you are working on a large project with version control, you can create R's equivalent of a virtual environment with the `packrat` package, such that anyone who wishes to build upon your code in the future will do so with the same packages and versions you used while developing the project.
```

Reply:
This introduces two new concepts, reproducibility in general and version control specifically. This does not really address the point the paragraph is making (that you have to load packages even if they are installed IMHO). I would not add this suggestion.

Reply:
Fair enough! I thought the idea of a `virtualenv` might click for people who are used to Python, but I can see how it is extraneous to this tutorial.